### PR TITLE
Help wanted, critic2 stable (new formula)

### PR DESCRIPTION
**Pull request by mkhorton**
Analysis of quantum chemical interactions in molecules and solids.

### ~~~Help wanted:~~~

~~~Formula is almost complete, I can get it to install, but I have to manually run `make` a second time ... the first time `make` is run, it gives a `BuildError: Failed executing: make` with:~~~

` USE Precision_Model, ONLY: stnd
    1
Fatal Error: Can't open module file 'precision_model.mod' for reading at (1): No such file or directory
compilation terminated.`

~~~For some reason, accessing the shell in interactive mode and running `make` a second time then works fine and the formula installs correctly and seems fully functional. I'm a bit confused as to what this issue is caused by.~~~

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
    - Note that author doesn't provide a version number, but this is a tagged release on github and marked as stable.
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?
      - ~~~Help wanted here! See above.~~~

### New formula: have you

- [x] Written a sensible test? (The best test is to compile and run a sample program.)
    - Test files are very large and not included in the source distribution, so test simply runs the program with an empty script to make sure it runs

### CODE REVIEW

File: critic2.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    end
+
+    if build.without? "libxc"
+      system "./configure", "--disable-debug",
+                            "--disable-dependency-tracking",
+                            "--disable-silent-rules",
+                            "--prefix=#{prefix}"
+    end
+
+    system "make"
```
Comment (jonchang) [inline critic2.rb]: > For some reason, accessing the shell in interactive mode and running make a second time then works fine and the formula installs correctly and seems fully functional. I'm a bit confused as to what this issue is caused by.

Try `ENV.deparallelize { system "make" }` as it's likely a race condition in the Makefile. Also consider notifying the upstream maintainers of this issue.

File: critic2.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class Critic2 < Formula
+  desc "Analysis of quantum chemical interactions in molecules and solids."
+  homepage "https://github.com/aoterodelaroza/critic2"
+  url "https://github.com/aoterodelaroza/critic2/archive/stable.tar.gz"
+  version "stable"
```
Comment (jonchang) [inline critic2.rb]: I don't think we can accept a formula without real version numbers:

`version (stable) is set to a string without a digit`

Could you please ask the upstream maintainers of critic2 to make a release with an actual version number?

File: critic2.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  sha256 "5c0803d47fbbb061b9f0b44e56eeebd7aa3298b275b9253fecdae7847a1ff903"
```
Comment (jonchang) [inline critic2.rb]: Include `# doi "..."` and `# tag "chemistry"` lines. See `lammps.rb` for an example.

File: critic2.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  option "with-libxc", "Compile with libxc support to calculate exchange and correlation densities"
+
+  depends_on "gcc" => :build
```
Comment (jonchang) [inline critic2.rb]: Are you sure this `depends_on "gcc"` and not `depends_on :fortran`?

File: critic2.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libxc" => :optional
+
+  def install
+    system "autoreconf", "-i"
+
+    if build.with? "libxc"
+      system "./configure", "--with-libxc-prefix=#{HOMEBREW_PREFIX}/lib",
+                            "--with-libxc-include=#{HOMEBREW_PREFIX}/include",
```
Comment (jonchang) [inline critic2.rb]: Do something like `--with-libxc-include=#{Formula["libxc"].opt_lib}"`

File: critic2.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                            "--disable-silent-rules",
+                            "--prefix=#{prefix}"
+    end
+
+    if build.without? "libxc"
+      system "./configure", "--disable-debug",
+                            "--disable-dependency-tracking",
+                            "--disable-silent-rules",
+                            "--prefix=#{prefix}"
+    end
```
Comment (jonchang) [inline critic2.rb]: Please refactor this section to remove duplication, by setting `args = [...]` then `if build.with? "libxc" args << "..."`. See `wcslib.rb` for an example of what I'm talking about.

File: critic2.rb (hunk 2 above)
Comment (mkhorton) [reply critic2.rb]: I raised this issue with the maintainers [here](https://github.com/aoterodelaroza/critic2/issues/6). They don't use minor version numbers in their code, instead using git for versioning, however this is a tagged stable release, so can be relied on to exist in future, and has an associated publication.

File: critic2.rb (hunk 3 above)
Comment (mkhorton) [reply critic2.rb]: Added these -- is there a list of tags homebrew-science uses? couldn't find a list in the wiki

File: critic2.rb (hunk 4 above)
Comment (mkhorton) [reply critic2.rb]: It was my understanding that gfortran has been rolled into gcc in homebrew, I'd tried "gfortran" as a dependency but it didn't work -- didn't realise `:fortran` could be used, thanks!

File: critic2.rb (hunk 5 above)
Comment (mkhorton) [reply critic2.rb]: Done

File: critic2.rb (hunk 6 above)
Comment (mkhorton) [reply critic2.rb]: Done

File: critic2.rb (hunk 1 above)
Comment (mkhorton) [reply critic2.rb]: This was exactly it! Thank you :)

**Comment by mkhorton** [comment]
Thanks @jonchang for the detailed review. I've pushed some changes, I think it should be okay now?

File: critic2.rb (hunk 4 above)
Comment (mkhorton) [reply critic2.rb]: Ah, I think the full gcc dependency is required after all...

**Comment by mkhorton** [comment]
(Looks like the only test failure now is "GitHub repository not notable enough" -- don't know what to say to this except that I find it useful! and the code has been in development since at least 2009 to the present day, and its associated papers have ~ 200 citations, so I think it's worth including! I don't think GitHub metrics are necessarily useful in this case.)

**Comment by mkhorton** [comment]
Hi @jonchang, I just wanted to follow up on this -- are there any additional changes I need to make for this to be accepted / have I missed anything? If I've understood the CI output correctly, the test failure is just notability, I believe it's fine otherwise.

**Comment by iMichka** [comment]
We updated our "new formula" policy for homebrew-science:

> We will no longer accept new formula in homebrew-science, as this tap is being phased out. Please consider submitting your formulae to https://github.com/Homebrew/homebrew-core or host it in your own tap (https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.html).

I am closing this pull request, if you need any advice please feel free to add a comment in this issue. Thanks!

See also #6365 for the discussion about what is going to happen with homebrew-science.

**Pull request closed without merging**